tardis/visualization/widgets/custom_abundance.py

```diff
@@ -0,0 +1,1700 @@
+        )
+        return np.add(str_symbols, str_mass)
+
+    def create_widgets(self):
+        """Create widget components in GUI and register callbacks for widgets."""
+        self.dpd_shell_no = ipw.Dropdown(
+            options=list(range(1, self.no_of_shells + 1)),
+            description="Shell No. ",
+            value=1,
+            layout=ipw.Layout(width="160px"),
+        )
+        self.dpd_shell_no.observe(self.dpd_shell_no_eventhandler, "value")
+        self.btn_prev = ipw.Button(
+            icon="chevron-left",
+            disabled=True,
+            layout=ipw.Layout(width="30px", height="30px"),
+        )
+        self.btn_prev.on_click(self.on_btn_prev)
+        self.btn_next = ipw.Button(
+            icon="chevron-right", layout=ipw.Layout(width="30px", height="30px")
+        )
+        self.btn_next.on_click(self.on_btn_next)
+
+        self.checks = [
+            ipw.Checkbox(
+                indent=False,
+                icon="lock",
+                layout=ipw.Layout(
+                    width="30px",
+                ),
+            )
+            for element in self.elements
+        ]
+        self.input_items = [
+            ipw.BoundedFloatText(min=0, max=1, step=0.01, description=element)
+            for element in self.elements
+        ]
+        for i in range(self.no_of_elements):
+            self.input_items[i].observe(self.input_item_eventhandler, "value")
+            self.input_items[i].index = i
+            self.checks[i].observe(self.check_eventhandler, "value")
+            self.checks[i].index = i
+
+        self.btn_norm = ipw.Button(
+            description="Normalize",
+            icon="cog",
+            layout=ipw.Layout(width="100px", margin="0 0 0 50px"),
+        )
+        self.btn_norm.on_click(self.on_btn_norm)
+        self.norm_warning = ipw.Valid(
+            value=False,
+            readout="Unnormalized",
+            style={"description_width": "initial"},
+            layout=ipw.Layout(visibility="hidden"),
+        )
+
+        self.symb_warning = ipw.Valid(
+            value=False, layout=ipw.Layout(visibility="hidden")
+        )
+        self.input_symb = ipw.Text(
+            description="Element: ",
+            style={"description_width": "initial"},
+            placeholder="symbol",
+            layout=ipw.Layout(width="125px"),
+        )
+        self.input_symb.observe(self.input_symb_eventhandler, "value")
+        self.btn_add_element = ipw.Button(
+            icon="plus-square",
+            description="Add",
+            disabled=True,
+            layout=ipw.Layout(width="60px"),
+        )
+        self.btn_add_element.on_click(self.on_btn_add_element)
+
+        # self.btn_apply = ipw.Button(
+        #     description=" Apply",
+        #     icon="pencil-square-o",
+        #     layout = ipw.Layout(width="100px")
+        # )
+        # self.btn_apply.on_click(self.on_btn_apply)
```

Comment:
is this needed?

Reply:
It's no longer needed. I'll delete this.